app/pages/resource/ResourcePage.js

```diff
@@ -1,39 +1,79 @@
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
+import { browserHistory } from 'react-router';
+import queryString from 'query-string';
 
-import SingleAvailabilityView from 'shared/availability-view/SingleAvailabilityView';
+import SelectableSingleAvailabilityView from 'shared/form-fields/reservation-time/SelectableSingleAvailabilityView';
 import ResourceDailyReportButton from 'shared/resource-daily-report-button';
 import ResourceInfo from './info';
 
-ResourcePage.propTypes = {
-  date: PropTypes.string.isRequired,
-  hideResourceImages: PropTypes.func.isRequired,
-  onDateChange: PropTypes.func.isRequired,
-  resource: PropTypes.object.isRequired,
-  resourceSearchUrl: PropTypes.string.isRequired,
-  showResourceImages: PropTypes.func.isRequired,
-  unit: PropTypes.object.isRequired,
-};
-export default function ResourcePage(props) {
-  return (
-    <div className="resource-page">
-      <ResourceInfo
-        hideResourceImages={props.hideResourceImages}
-        resource={props.resource}
-        resourceSearchUrl={props.resourceSearchUrl}
-        showResourceImages={props.showResourceImages}
-        unit={props.unit}
-      />
-      <h3>Varaustilanne</h3>
-      <p className="help-text">Klikkaa vapaata aikaa varauksen aloittamiseksi</p>
-      <SingleAvailabilityView
-        date={props.date}
-        onDateChange={props.onDateChange}
-        resource={props.resource.id}
-      />
-      <ResourceDailyReportButton
-        resourceIds={[props.resource.id]}
-        date={props.date}
-      />
-    </div>
-  );
+export default class ResourcePage extends Component {
+  static propTypes = {
+    date: PropTypes.string.isRequired,
+    hideResourceImages: PropTypes.func.isRequired,
+    onDateChange: PropTypes.func.isRequired,
+    resource: PropTypes.object.isRequired,
+    resourceSearchUrl: PropTypes.string.isRequired,
+    showResourceImages: PropTypes.func.isRequired,
+    unit: PropTypes.object.isRequired,
+  };
+
+  state = {
+    value: {
```

Comment:
- [ ] In the context of ResourcePage state it might make more sense to use some other name than `value`, for example `timelineSelection` or something like that.